.dockerignore

```diff
@@ -0,0 +1,10 @@
+/.dockerignore
+/.gitignore
+/.git
+/archive
+/build
+/common/*.mk
+/*.md
+/Jenkinsfile
+/Makefile
+/Makefile.win
```

Comment:
FWIW, with BuildKit using a session to send only what's needed, and with the Dockerfiles already being specific about what to `COPY`, I later realised we probably don't strictly need the `.dockerignore` (guess it doesn't really hurt to have one, but I can remove if we think it's not needed)